### PR TITLE
Fix c3dfileadapters docs

### PR DESCRIPTION
**Pull request by aymanhab**
Fixes issue #2673 partially

### Brief summary of changes
Update code snippets for python, Matlab, Java

### CHANGELOG.md (choose one)

- no need to update because documentation fix

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2676)

### CODE REVIEW

**Comment by aymanhab** [comment]
@tkuchida sure look forward to your feedback.
@jenhicks Any suggestions regarding updating confluence? Are the pages referred to in issue #2673 shared by 4.0 and 4.1 (which have different APIs)? Thanks

File: OpenSim/Common/C3DFileAdapter.h [hunk 1]
```diff
@@ -68,29 +68,37 @@ class OSIMCOMMON_API C3DFileAdapter : public FileAdapter {
...
+        TimeSeriesTableVec3 forcesTable = c3dFileAdapter.getForcesTable(tables);
+
         \endcode
 
         <b>MATLAB example</b>
         \code{.m}
          C3DFileAdapter c3dFileAdapter;
          c3dFileAdapter.setLocationForForceExpression(1);
-        tables = C3DFileAdapter.read("myData.c3d");
+         tables = C3DFileAdapter.read("myData.c3d");
```
Comment (tkuchida) [inline OpenSim/Common/C3DFileAdapter.h]: When did Matlab switch to double quotes? What version are you using?

File: OpenSim/Common/C3DFileAdapter.h [hunk 2]
```diff
@@ -68,29 +68,37 @@ class OSIMCOMMON_API C3DFileAdapter : public FileAdapter {
...
         \endcode
 
         <b>Python example</b>
         \code{.py}
         import opensim
-        C3DFileAdapter c3dFileAdapter;
+        c3dFileAdapter = opensim.C3DFileAdapter()
         c3dFileAdapter.setLocationForForceExpression(opensim.C3DFileAdapter.ForceLocation_CenterOfPressure);
         tables = c3dFileAdapter.read("myData.c3d")
+        markersTable = c3dFileAdapter.getMarkersTable(tables)
```
Comment (tkuchida) [inline OpenSim/Common/C3DFileAdapter.h]: I'll have to upgrade to 4.1 to say for sure, but even in 4.0 there's a `read()` method that takes a string as an argument. Running `c3dFileAdapter.read('myData.c3d')` in Spyder gives the following error:
```
NotImplementedError: Wrong number or type of arguments for overloaded function 'C3DFileAdapter_read'.
  Possible C/C++ prototypes are:
    OpenSim::C3DFileAdapter::read(std::string const &,OpenSim::C3DFileAdapter::ForceLocation)
    OpenSim::C3DFileAdapter::read(std::string const &)
    OpenSim::C3DFileAdapter::read(std::string const &,unsigned int)
```
I haven't figured this one out. Do I also need to upgrade my Python version to run this? What version of Python are you using?

Comment (aymanhab) [reply OpenSim/Common/C3DFileAdapter.h]: @tkuchida Thanks for checking, yes you need to upgrade to 4.1 and Python 3.7+ (Python 2 reached end of life on 1/1/20). Method 'read' existed but was not implemented in the interface in 4.0, now it is implemented in 4.1 without additional arguments as it overrides base class signature.

File: OpenSim/Common/C3DFileAdapter.h (hunk 1 above)
Comment (aymanhab) [reply OpenSim/Common/C3DFileAdapter.h]: Thanks for the catch @tkuchida, I didn't change the line but I believe you're correct. I made the change accordingly.

**Comment by jenhicks** [comment]
The confluence documentation is shared by 4.0 and 4.1. So you could just have two sections ... one for 4.0 and one for 4.1. If you get the content up, I can clean up any formatting issues.

**Comment by aymanhab** [comment]
Pending review/verification with python in 4.1 @chrisdembia @carmichaelong or @nickbianco Any volunteers 😃

File: OpenSim/Common/C3DFileAdapter.h (hunk 2 above)
Comment (tkuchida) [reply OpenSim/Common/C3DFileAdapter.h]: Thanks. Might want to add a note that this code won't work in 4.0.

File: OpenSim/Common/C3DFileAdapter.h (hunk 1 above)
Comment (tkuchida) [reply OpenSim/Common/C3DFileAdapter.h]: Ah, ok thanks. I don't have Matlab hooked up but maybe a Matlab user could test this.

**Comment by carmichaelong** [comment]
@aymanhab I could test but it might not be until early next week (small chance it could happen by the end of the week). Are there any specific time constraints with this PR?

**Comment by aymanhab** [comment]
Thanks @carmichaelong, absolutely no time constraints we just want the documentation to be updated correctly.

**Comment by carmichaelong** [comment]
Tested using `test_walking.c3d` file in the MATLAB testing resources folder.

Example code worked with Python as is (with just a file name change), without any errors/warning.

I needed to change the MATLAB code to the following:
```
import org.opensim.modeling.*;
c3dFileAdapter = C3DFileAdapter();
c3dFileAdapter.setLocationForForceExpression(1);
tables = c3dFileAdapter.read("test_walking.c3d");
markersTable = c3dFileAdapter.getMarkersTable(tables);
forcesTable = c3dFileAdapter.getForcesTable(tables);
```
This worked, but threw a warning from BTK:
```
[BTK WARNING] Origin for the force platform #1 seems to be located from the center of the working surface instead of the inverse. Data are inverted to locate the center of the working surface from the platform's origin.
[BTK WARNING] Origin for the force platform #1 seems to be located from the center of the working surface instead of the inverse. Data are inverted to locate the center of the working surface from the platform's origin.
```

**Comment by aymanhab** [comment]
Thanks @carmichaelong, much appreciated. I updated the file per your feedback, if it looks good feel free to merge.

**Comment by carmichaelong** [comment]
Oops, I just realized I used double quotes in the MATLAB example, which did work in 2019a. Perhaps due to the previous discussion that line should be single quotes instead?

**Comment by aymanhab** [comment]
Thanks @carmichaelong, updated and merged.

**Pull request merged**